platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3743,13 +3747,34 @@
         */
         $scope.addEntityInstance = function () {
             blockUI();
-
-            var values = $scope.currentRecord.fields;
+            $scope.addEntityInstanceBase($scope.currentRecord);
+        };
+        $scope.addEntityInstanceBase = function (inst)
+        {
+            var values = inst.fields;
             angular.forEach (values, function(value, key) {
                 value.value = value.value === 'null' ? null : value.value;
             });
+            for(var i in inst.fields) {
+                var fd = inst.fields[i];
+
+                // {readonly, uiFilterable, . . .} = fields(FieldDto) - fields(FieldRecord)
+                delete fd["readOnly"];
+                delete fd["uiFilterable"];
+                delete fd["uiChanged"];
+                delete fd["lookups"];
+
+                // TODO: where is "discriminated" coming from?  Not isDiscriminated?
+                var entitiesDerived = fd.entitiesDerived;
+                if(entitiesDerived) {
+                    for (var j in entitiesDerived) {
```

Comment:
Generally speaking this construct should be avoided: http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-such-a-bad-idea

I can see this iterates over indexes for you? I believe that depenends on the browser.